### DISCUSSION TITLE
Can i run  a GUI app in wasmer

### DISCUSSION THREAD

**Discussion by subham008** [wasmerio/wasmer#5122] [General]
means can it open a window and show  Graphics , am using slint  with Rust to develop GUI app , i compiles it with target wasm-wasi ,  it compiled success  , but it failed to run  via wasmer 

Here is the error it throw

cannot call wasm-bindgen imported functions on non-wasm targets
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: RuntimeError: unreachable
    at __rust_start_panic (CloclApp-5b857290c8239c5c.wasm[40399]:0xe6c379)
    at rust_panic (CloclApp-5b857290c8239c5c.wasm[40391]:0xe6bffa)
    at std::panicking::rust_panic_with_hook::hd3fb69bc0aea298a (CloclApp-5b857290c8239c5c.wasm[40390]:0xe6bf33)
    at std::panicking::begin_panic::{{closure}}::h82a3140b1f5226fa (CloclApp-5b857290c8239c5c.wasm[21094]:0x6268d6)
    at std::sys_common::backtrace::__rust_end_short_backtrace::h389c2159a3617d66 (CloclApp-5b857290c8239c5c.wasm[21093]:0x62681b)
    at std::panicking::begin_panic::he0a82a11bc645100 (CloclApp-5b857290c8239c5c.wasm[21101]:0x626e7e)
    at js_sys::global::get_global_object::Global::get_self::__wbg_self_05040bd9523805b9::hd7f24c7f9240b9db (CloclApp-5b857290c8239c5c.wasm[21021]:0x624763)
    at js_sys::global::get_global_object::Global::get_self::h38c0ee053c321fa9 (CloclApp-5b857290c8239c5c.wasm[20981]:0x6230cf)
    at js_sys::global::get_global_object::h59ef74c555dee57a (CloclApp-5b857290c8239c5c.wasm[20980]:0x622d5b)
    at js_sys::global::GLOBAL::__init::h35870c0c98c8840e (CloclApp-5b857290c8239c5c.wasm[21017]:0x624504)
    at js_sys::global::GLOBAL::__getit::{{closure}}::hce1ed8d060196183 (CloclApp-5b857290c8239c5c.wasm[21019]:0x62463c)
    at std::sys::thread_local::lazy::LazyKeyInner<T>::initialize::hbda2b16f17bdb7d7 (CloclApp-5b857290c8239c5c.wasm[21096]:0x626ae3)
    at std::sys::thread_local::static_local::Key<T>::get::hadb9c27ff7957c18 (CloclApp-5b857290c8239c5c.wasm[21095]:0x6269c5)
    at js_sys::global::GLOBAL::__getit::ha6fe8295cc58671d (CloclApp-5b857290c8239c5c.wasm[21018]:0x624550)
    at std::thread::local::LocalKey<T>::try_with::hfd927d8e487a6499 (CloclApp-5b857290c8239c5c.wasm[21100]:0x626d38)
    at std::thread::local::LocalKey<T>::with::h35d15b1239fbb731 (CloclApp-5b857290c8239c5c.wasm[21099]:0x626c6f)
    at js_sys::global::h09273248fd3e8115 (CloclApp-5b857290c8239c5c.wasm[20977]:0x622c7a)
    at web_sys::window::he7766aef383e883d (CloclApp-5b857290c8239c5c.wasm[20923]:0x620ab4)
    at winit::platform_impl::platform::event_loop::runner::Shared::new::hc9d6c1c4e89ff417 (CloclApp-5b857290c8239c5c.wasm[12608]:0x427e70)
    at winit::platform_impl::platform::event_loop::window_target::EventLoopWindowTarget<T>::new::hbdf68031984bc246 (CloclApp-5b857290c8239c5c.wasm[3398]:0xfc9cd)
    at winit::platform_impl::platform::event_loop::EventLoop<T>::new::hd54c17ac9d927c92 (CloclApp-5b857290c8239c5c.wasm[1871]:0xa8124)
    at winit::event_loop::EventLoopBuilder<T>::build::hf7727f9553d34307 (CloclApp-5b857290c8239c5c.wasm[3091]:0xe8a05)
    at i_slint_backend_winit::event_loop::NotRunningEventLoop::new::h6d22e1b05ab044ff (CloclApp-5b857290c8239c5c.wasm[2067]:0xb725a)
    at i_slint_backend_winit::event_loop::with_window_target::{{closure}}::h4470db5487d177bb (CloclApp-5b857290c8239c5c.wasm[2097]:0xb8d1e)
    at std::thread::local::LocalKey<T>::try_with::hee6be1adcdd04c77 (CloclApp-5b857290c8239c5c.wasm[3293]:0xf72e5)
    at std::thread::local::LocalKey<T>::with::hf278ddbaa5982c0d (CloclApp-5b857290c8239c5c.wasm[3292]:0xf7175)
    at i_slint_backend_winit::event_loop::with_window_target::h32e4a7f9f8d42a32 (CloclApp-5b857290c8239c5c.wasm[2085]:0xb8461)
    at i_slint_backend_winit::Backend::new_with_renderer_by_name::h2d34d469669321db (CloclApp-5b857290c8239c5c.wasm[1182]:0x67e6f)
    at i_slint_backend_winit::Backend::new::h97006b9ced94618c (CloclApp-5b857290c8239c5c.wasm[1181]:0x67e03)
    at i_slint_backend_selector::create_winit_backend::h5e25265e876bec4c (CloclApp-5b857290c8239c5c.wasm[974]:0x3ca8c)
    at i_slint_backend_selector::create_default_backend::h911b327432ef67cf (CloclApp-5b857290c8239c5c.wasm[975]:0x3d2a3)
    at i_slint_backend_selector::create_backend::ha6d8cc1267cd72dc (CloclApp-5b857290c8239c5c.wasm[976]:0x3dd41)
    at core::ops::function::FnOnce::call_once::he653d61323971b95 (CloclApp-5b857290c8239c5c.wasm[855]:0x33385)
    at i_slint_core::context::with_platform::{{closure}}::hb9a6970aeeb85f12 (CloclApp-5b857290c8239c5c.wasm[862]:0x33ee9)
    at std::thread::local::LocalKey<T>::try_with::h8e25a995f0cdde4b (CloclApp-5b857290c8239c5c.wasm[868]:0x34445)
    at std::thread::local::LocalKey<T>::with::h4ce87dc619e34b65 (CloclApp-5b857290c8239c5c.wasm[852]:0x33121)
    at i_slint_core::context::with_platform::h57e105d58d99e91c (CloclApp-5b857290c8239c5c.wasm[851]:0x330ad)
    at i_slint_backend_selector::with_platform::h157c23a811ab1e2e (CloclApp-5b857290c8239c5c.wasm[865]:0x342d1)
    at slint::private_unstable_api::ensure_backend::h788794ceba301e0b (CloclApp-5b857290c8239c5c.wasm[873]:0x34779)
    at CloclApp::slint_generatedMainWindow::InnerMainWindow::new::h1547740d0e4d9b9c (CloclApp-5b857290c8239c5c.wasm[35]:0x11139)
    at CloclApp::slint_generatedMainWindow::MainWindow::new::hc634493655f35d42 (CloclApp-5b857290c8239c5c.wasm[57]:0x12e80)
    at CloclApp::main::h9e25983ed8ff530f (CloclApp-5b857290c8239c5c.wasm[541]:0x26e45)
    at core::ops::function::FnOnce::call_once::h9c5f76c3609afcd5 (CloclApp-5b857290c8239c5c.wasm[642]:0x2be66)
    at std::sys_common::backtrace::__rust_begin_short_backtrace::h7ff92d0c83189c36 (CloclApp-5b857290c8239c5c.wasm[466]:0x21cc3)
    at std::rt::lang_start::{{closure}}::hdda42f9216d355d7 (CloclApp-5b857290c8239c5c.wasm[842]:0x32e1e)
    at std::rt::lang_start_internal::h8f61396649e22e0e (CloclApp-5b857290c8239c5c.wasm[40323]:0xe6771c)
    at std::rt::lang_start::hd234a4c48a1e88cf (CloclApp-5b857290c8239c5c.wasm[841]:0x32dba)
    at __main_void (CloclApp-5b857290c8239c5c.wasm[542]:0x27031)
    at _start (CloclApp-5b857290c8239c5c.wasm[15]:0xc94b)
╰─▶ 1: RuntimeError: unreachable
           at __rust_start_panic (CloclApp-5b857290c8239c5c.wasm[40399]:0xe6c379)
           at rust_panic (CloclApp-5b857290c8239c5c.wasm[40391]:0xe6bffa)
           at std::panicking::rust_panic_with_hook::hd3fb69bc0aea298a (CloclApp-5b857290c8239c5c.wasm[40390]:0xe6bf33)
           at std::panicking::begin_panic::{{closure}}::h82a3140b1f5226fa (CloclApp-5b857290c8239c5c.wasm[21094]:0x6268d6)
           at std::sys_common::backtrace::__rust_end_short_backtrace::h389c2159a3617d66 (CloclApp-5b857290c8239c5c.wasm[21093]:0x62681b)
           at std::panicking::begin_panic::he0a82a11bc645100 (CloclApp-5b857290c8239c5c.wasm[21101]:0x626e7e)
           at js_sys::global::get_global_object::Global::get_self::__wbg_self_05040bd9523805b9::hd7f24c7f9240b9db (CloclApp-5b857290c8239c5c.wasm[21021]:0x624763)
           at js_sys::global::get_global_object::Global::get_self::h38c0ee053c321fa9 (CloclApp-5b857290c8239c5c.wasm[20981]:0x6230cf)
           at js_sys::global::get_global_object::h59ef74c555dee57a (CloclApp-5b857290c8239c5c.wasm[20980]:0x622d5b)
           at js_sys::global::GLOBAL::__init::h35870c0c98c8840e (CloclApp-5b857290c8239c5c.wasm[21017]:0x624504)
           at js_sys::global::GLOBAL::__getit::{{closure}}::hce1ed8d060196183 (CloclApp-5b857290c8239c5c.wasm[21019]:0x62463c)
           at std::sys::thread_local::lazy::LazyKeyInner<T>::initialize::hbda2b16f17bdb7d7 (CloclApp-5b857290c8239c5c.wasm[21096]:0x626ae3)
           at std::sys::thread_local::static_local::Key<T>::get::hadb9c27ff7957c18 (CloclApp-5b857290c8239c5c.wasm[21095]:0x6269c5)
           at js_sys::global::GLOBAL::__getit::ha6fe8295cc58671d (CloclApp-5b857290c8239c5c.wasm[21018]:0x624550)
           at std::thread::local::LocalKey<T>::try_with::hfd927d8e487a6499 (CloclApp-5b857290c8239c5c.wasm[21100]:0x626d38)
           at std::thread::local::LocalKey<T>::with::h35d15b1239fbb731 (CloclApp-5b857290c8239c5c.wasm[21099]:0x626c6f)
           at js_sys::global::h09273248fd3e8115 (CloclApp-5b857290c8239c5c.wasm[20977]:0x622c7a)
           at web_sys::window::he7766aef383e883d (CloclApp-5b857290c8239c5c.wasm[20923]:0x620ab4)
           at winit::platform_impl::platform::event_loop::runner::Shared::new::hc9d6c1c4e89ff417 (CloclApp-5b857290c8239c5c.wasm[12608]:0x427e70)
           at winit::platform_impl::platform::event_loop::window_target::EventLoopWindowTarget<T>::new::hbdf68031984bc246 (CloclApp-5b857290c8239c5c.wasm[3398]:0xfc9cd)
           at winit::platform_impl::platform::event_loop::EventLoop<T>::new::hd54c17ac9d927c92 (CloclApp-5b857290c8239c5c.wasm[1871]:0xa8124)
           at winit::event_loop::EventLoopBuilder<T>::build::hf7727f9553d34307 (CloclApp-5b857290c8239c5c.wasm[3091]:0xe8a05)
           at i_slint_backend_winit::event_loop::NotRunningEventLoop::new::h6d22e1b05ab044ff (CloclApp-5b857290c8239c5c.wasm[2067]:0xb725a)
           at i_slint_backend_winit::event_loop::with_window_target::{{closure}}::h4470db5487d177bb (CloclApp-5b857290c8239c5c.wasm[2097]:0xb8d1e)
           at std::thread::local::LocalKey<T>::try_with::hee6be1adcdd04c77 (CloclApp-5b857290c8239c5c.wasm[3293]:0xf72e5)
           at std::thread::local::LocalKey<T>::with::hf278ddbaa5982c0d (CloclApp-5b857290c8239c5c.wasm[3292]:0xf7175)
           at i_slint_backend_winit::event_loop::with_window_target::h32e4a7f9f8d42a32 (CloclApp-5b857290c8239c5c.wasm[2085]:0xb8461)
           at i_slint_backend_winit::Backend::new_with_renderer_by_name::h2d34d469669321db (CloclApp-5b857290c8239c5c.wasm[1182]:0x67e6f)
           at i_slint_backend_winit::Backend::new::h97006b9ced94618c (CloclApp-5b857290c8239c5c.wasm[1181]:0x67e03)
           at i_slint_backend_selector::create_winit_backend::h5e25265e876bec4c (CloclApp-5b857290c8239c5c.wasm[974]:0x3ca8c)
           at i_slint_backend_selector::create_default_backend::h911b327432ef67cf (CloclApp-5b857290c8239c5c.wasm[975]:0x3d2a3)
           at i_slint_backend_selector::create_backend::ha6d8cc1267cd72dc (CloclApp-5b857290c8239c5c.wasm[976]:0x3dd41)
           at core::ops::function::FnOnce::call_once::he653d61323971b95 (CloclApp-5b857290c8239c5c.wasm[855]:0x33385)
           at i_slint_core::context::with_platform::{{closure}}::hb9a6970aeeb85f12 (CloclApp-5b857290c8239c5c.wasm[862]:0x33ee9)
           at std::thread::local::LocalKey<T>::try_with::h8e25a995f0cdde4b (CloclApp-5b857290c8239c5c.wasm[868]:0x34445)
           at std::thread::local::LocalKey<T>::with::h4ce87dc619e34b65 (CloclApp-5b857290c8239c5c.wasm[852]:0x33121)
           at i_slint_core::context::with_platform::h57e105d58d99e91c (CloclApp-5b857290c8239c5c.wasm[851]:0x330ad)
           at i_slint_backend_selector::with_platform::h157c23a811ab1e2e (CloclApp-5b857290c8239c5c.wasm[865]:0x342d1)
           at slint::private_unstable_api::ensure_backend::h788794ceba301e0b (CloclApp-5b857290c8239c5c.wasm[873]:0x34779)
           at CloclApp::slint_generatedMainWindow::InnerMainWindow::new::h1547740d0e4d9b9c (CloclApp-5b857290c8239c5c.wasm[35]:0x11139)
           at CloclApp::slint_generatedMainWindow::MainWindow::new::hc634493655f35d42 (CloclApp-5b857290c8239c5c.wasm[57]:0x12e80)
           at CloclApp::main::h9e25983ed8ff530f (CloclApp-5b857290c8239c5c.wasm[541]:0x26e45)
           at core::ops::function::FnOnce::call_once::h9c5f76c3609afcd5 (CloclApp-5b857290c8239c5c.wasm[642]:0x2be66)
           at std::sys_common::backtrace::__rust_begin_short_backtrace::h7ff92d0c83189c36 (CloclApp-5b857290c8239c5c.wasm[466]:0x21cc3)
           at std::rt::lang_start::{{closure}}::hdda42f9216d355d7 (CloclApp-5b857290c8239c5c.wasm[842]:0x32e1e)
           at std::rt::lang_start_internal::h8f61396649e22e0e (CloclApp-5b857290c8239c5c.wasm[40323]:0xe6771c)
           at std::rt::lang_start::hd234a4c48a1e88cf (CloclApp-5b857290c8239c5c.wasm[841]:0x32dba)
           at __main_void (CloclApp-5b857290c8239c5c.wasm[542]:0x27031)
           at _start (CloclApp-5b857290c8239c5c.wasm[15]:0xc94b)

**Comment by Arshia001** [comment]
Hey!

Wasmer does not support GUIs out of the box. There are two ways you can use WASM to do UI:

* Loading and running a WASM module in the browser or an electron app, which I guess is not what you're trying to do, or
* Have bindings to the host OS, similar to WASIX, which let WASM modules call into the OS to create windows and draw things.

The second approach isn't implemented in Wasmer (or any other WASM runtime I know of), primarily because there is no standard way to render GUI that works across all operating systems, so it's very difficult to standardize an interface that runtimes can implement.

TL;DR no, Wasmer does not support GUI apps at the moment.
